protocol/avdl/keybase1/saltpack.avdl

```diff
@@ -69,19 +69,38 @@ protocol saltpack {
   }
 
   string saltpackEncryptString(int sessionID, string plaintext, SaltpackFrontendEncryptOptions opts);
+  string saltpackEncryptFile(int sessionID, string filename, SaltpackFrontendEncryptOptions opts);
 
   record SaltpackPlaintextResult {
     SaltpackEncryptedMessageInfo info;
     string plaintext;
+    boolean signed;
   }
   SaltpackPlaintextResult saltpackDecryptString(int sessionID, string ciphertext);
 
+  record SaltpackFileResult {
+    SaltpackEncryptedMessageInfo info;
+    string decryptedFilename;
+    boolean signed;
+  }
+  SaltpackFileResult saltpackDecryptFile(int sessionID, string encryptedFilename);
+
   string saltpackSignString(int sessionID, string plaintext);
+  string saltpackSignFile(int sessionID, string filename);
 
   record SaltpackVerifyResult {
     KID signingKID;
     SaltpackSender sender;
     string plaintext;
+    boolean verified;
   }
   SaltpackVerifyResult saltpackVerifyString(int sessionID, string signedMsg);
+
+  record SaltpackVerifyFileResult {
+    KID signingKID;
```

Comment:
this value looks like it's optional coming out of saltpack. should it also be optional here so we don't pass empty strings to the frontend rather than possibly-easier-to-handle nils? and same for sender. and same for SaltpackVerifyResult.

Reply:
I believe that the way this is hooked up you will either get a valid SaltpackVerify*Result or an error.  I will double-check.

Reply:
i'm also fine with throwing an error if it's nil. then we'll find the issue when it happens.